src/ra_wm_handler.cpp

```diff
@@ -0,0 +1,227 @@
+#include "ra_wm_handler.h"
+
+#include <memory>
+#include <CommCtrl.h>
+#include <WindowsX.h>
+
+namespace ra {
+namespace ui {
+
+_Use_decl_annotations_
+BOOL WM_Handler::OnInitDialog(HWND hwnd, HWND hwndFocus, LPARAM lParam) noexcept
+{
+    ::SetFocus(hwndFocus);
+    // If it returns nonzero, the focus will go back to hwnd
+    return ::SetWindowLongPtr(hwnd, DWLP_USER, lParam);
+}
+
+_Use_decl_annotations_
+void WM_Handler::OnCommand(HWND hwnd, int id, HWND hwndCtl, UINT codeNotify) noexcept
+{
+    return FORWARD_WM_COMMAND(hwnd, id, hwndCtl, codeNotify, SendMessage);
```

Comment:
Be careful with these "`Send/PostMessage`" defaults, they shouldn't be called directly in the Proc. However, it should be fine to use them while processing messages for a different window. Should these be protected?